### PR TITLE
Provide an accurate estimate of mblock allocated length property

### PR DESCRIPTION
**Pull request by nabeelmmd**
Signed-off-by: Nabeel M Mohamed <50154757+nabeelmmd@users.noreply.github.com>

## Issue(s) Addressed
NFSE-4938

### CODE REVIEW

File: lib/mpool/src/mblock_file.c [hunk 1]
```diff
@@ -92,7 +92,7 @@ struct mblock_file {
     int            fileid;
     int            fd;
 
-    atomic_int *wlenv;
+    atomic_uint *wlenv;
```
Comment (tristan957) [inline lib/mpool/src/mblock_file.c:95]: Would rather this be based on fixed size integers. Seems like you use uint32_t later. Maybe `atomic_uint_least32_t`

File: lib/mpool/src/mblock_file.c [hunk 2]
```diff
@@ -581,6 +581,43 @@ mblock_file_meta_log(struct mblock_file *mbfp, uint64_t *mbidv, int mbidc, bool
...
+static HSE_ALWAYS_INLINE uint32_t
+mblock_wlen_get(struct mblock_file *mbfp, uint64_t mbid)
+{
+    return atomic_read(mbfp->wlenv + block_id(mbid)) & MBLOCK_WLEN_PREALLOC_MASK;
+}
+
+static HSE_ALWAYS_INLINE bool
+mblock_is_prealloc(struct mblock_file *mbfp, uint64_t mbid)
+{
+    uint32_t wlen = atomic_read(mbfp->wlenv + block_id(mbid));
```
Comment (tristan957) [inline lib/mpool/src/mblock_file.c:607]: const

File: lib/mpool/src/mblock_file.c [hunk 3]
```diff
@@ -581,6 +581,43 @@ mblock_file_meta_log(struct mblock_file *mbfp, uint64_t *mbidv, int mbidc, bool
...
+    if (prealloc)
+        wlen |= (1U << MBLOCK_WLEN_PREALLOC_SHIFT);
+
+    atomic_set(mbfp->wlenv + block_id(mbid), wlen);
+
+    return 0;
+}
+
+static HSE_ALWAYS_INLINE uint32_t
+mblock_wlen_get(struct mblock_file *mbfp, uint64_t mbid)
```
Comment (tristan957) [inline lib/mpool/src/mblock_file.c:599]: const parameters for getters!

File: lib/mpool/src/mblock_file.c [hunk 4]
```diff
@@ -581,6 +581,43 @@ mblock_file_meta_log(struct mblock_file *mbfp, uint64_t *mbidv, int mbidc, bool
...
+{
+    return atomic_read(mbfp->wlenv + block_id(mbid)) & MBLOCK_WLEN_PREALLOC_MASK;
+}
+
+static HSE_ALWAYS_INLINE bool
+mblock_is_prealloc(struct mblock_file *mbfp, uint64_t mbid)
+{
+    uint32_t wlen = atomic_read(mbfp->wlenv + block_id(mbid));
+
+    return ((wlen >> MBLOCK_WLEN_PREALLOC_SHIFT) == 1);
```
Comment (tristan957) [inline lib/mpool/src/mblock_file.c:609]: Extra set of parens

File: lib/mpool/src/mblock_file.c [hunk 5]
```diff
@@ -1303,7 +1320,7 @@ mblock_file_info_get(struct mblock_file *mbfp, struct mblock_file_info *info)
 }
 
 merr_t
-mblock_file_mbinfo_get(struct mblock_file *mbfp, uint64_t mbid, struct mblock_file_mbinfo *mbinfo)
+mblock_info_get(struct mblock_file *mbfp, uint64_t mbid, struct mblock_file_mbinfo *mbinfo)
```
Comment (tristan957) [inline lib/mpool/src/mblock_file.c:1323]: const parameters where possible

File: tests/unit/mpool/mcache_test.c [hunk 6]
```diff
@@ -287,13 +287,13 @@ MTF_DEFINE_UTEST_PREPOST(mcache_test, mcache_invalid_args, mpool_test_pre, mpool
     err = mblock_fset_unmap(mbfsp, 0xffffffff);
     ASSERT_EQ(EINVAL, merr_errno(err));
 
-    err = mblock_file_map_getbase(NULL, mbid, &addr, NULL);
+    err = mblock_map_getbase(NULL, mbid, &addr, NULL);
```
Comment (tristan957) [inline tests/unit/mpool/mcache_test.c:290]: If we are renaming this function, maybe put an underscore between get and base

**Review by tristan957 — APPROVED** [review]

File: lib/mpool/src/mblock_file.c (hunk 1 above)
Comment (nabeelmmd) [reply lib/mpool/src/mblock_file.c]: Fixed, thnx!

File: lib/mpool/src/mblock_fset.c [hunk 7]
```diff
@@ -870,7 +870,7 @@ mblock_fset_clone(
     if (err)
         goto errout;
 
-    err = mblock_file_wlen_set(tgt_mbfp, tgt_mbid, off + len);
+    err = mblock_wlen_set(tgt_mbfp, tgt_mbid, off + len, 0);
```
Comment (alexttx) [inline lib/mpool/src/mblock_fset.c:873]: Change last param from `0` to `false`.

Also, is it possible at this call site for `off+len` to exceed `tgt_mbfp->mblocksz`?

If not, then consider changing `mblock_wlen_set` to return `void` and change the param checks to `INVARIANT` macros.

**Review by alexttx — APPROVED** [review]

File: lib/mpool/src/mblock_fset.c (hunk 7 above)
Comment (nabeelmmd) [reply lib/mpool/src/mblock_fset.c]: Done, thnx!

**Pull request merged**